xpk.py

```diff
@@ -1877,23 +2130,176 @@ def create_cluster_network_config(args) -> int:
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    Tuple with string with the system characteristics and
+    int of 0 if successful and 1 otherwise.
+  """
+  capacity_type = CapacityType.UNKNOWN
+  num_types = 0
+  return_code = 0
+
+  # Determine the capacity argument.
+  if args.on_demand:
+    capacity_type = CapacityType.ON_DEMAND
+    num_types+=1
+  if args.reservation:
+    return_code = verify_reservation_exists(args)
+    if return_code > 0:
+      return capacity_type, return_code
+    capacity_type = CapacityType.RESERVATION
+    num_types+=1
+  if args.spot:
+    capacity_type = CapacityType.SPOT
+    num_types+=1
+
+  # Check that the number of user arguments provided is valid.
+  if num_types == 0:
+    capacity_type = CapacityType.UNKNOWN
+  elif num_types != 1:
+    xpk_print(
+      'ERROR: User specified more than one of the following arguments. Please'
+      ' specify only one of `--reservation=$RESERVATION_NAME`, `--on-demand`'
+      ' or `--spot`.'
+    )
+    return_code = 1
+
+  return capacity_type, return_code
+
+
+def get_capacity_arguments_from_capacity_type(args, capacity_type: CapacityType) -> tuple[str, int]:
+  """Determine the TPU Nodepool creation capacity arguments needed.
+
+  Args:
+    args: user provided arguments for running the command.
+    capacity_type: The type of capacity the user configured.
+
+  Returns:
+    Tuple with string with the capacity argument to use and
+    int of 0 if successful and 1 otherwise.
+  """
+  capacity_args = ""
+  return_code = 0
+
+  match capacity_type:
+    case CapacityType.ON_DEMAND:
+      capacity_args = ""
+    case CapacityType.SPOT:
+      capacity_args = "--spot"
+    case CapacityType.RESERVATION:
+      capacity_args = (
+        f'--reservation-affinity=specific --reservation={args.reservation}'
+      )
+    case _:
+      xpk_print(f'Unknown capacity type: {capacity_type}. Unable to determine capacity args.')
+      return_code = 1
+  return capacity_args, return_code
+
+
+def get_capacity_node_selectors_from_capacity_type(args, capacity_type: str) -> tuple[str, int]:
+  """Determine the node selectors for a workload to run on a specific capacity type.
+
+  Args:
+    args: user provided arguments for running the command.
+    capacity_type: The type of capacity the user configured.
+
+  Returns:
+    Tuple with string with the node selectors to use and
+    int of 0 if successful and 1 otherwise.
+  """
+  node_selector = ""
+  return_code = 0
+
+  match capacity_type:
+    case CapacityType.ON_DEMAND.name:
+      node_selector = ""
+    case CapacityType.SPOT.name:
+      node_selector = 'cloud.google.com/gke-spot=\"true\"'
+    case CapacityType.RESERVATION.name:
+      node_selector = (
+        f'cloud.google.com/reservation-name: {args.reservation}'
+      )
+    case _:
+      xpk_print(f'Unknown capacity type: {capacity_type}. Unable to determine capacity args.')
```

Comment:
nit: `unable to determine the node selectors`

Reply:
Thank you Surbhi!!